codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -287,7 +287,7 @@ public ChannelFuture writePushPromise(ChannelHandlerContext ctx, int streamId, i
     @Override
     public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
             ChannelPromise promise) {
-        return lifecycleManager.writeGoAway(ctx, lastStreamId, errorCode, debugData, promise);
+        return lifecycleManager.closeConnectionRemote(ctx, lastStreamId, errorCode, debugData, promise);
```

Comment:
This seems to be the opposite logic for `closeStreamRemote`.  Writing a GOAWAY should be closing the local side, shouldn't it?